### DISCUSSION TITLE
igit size `nuxt/nuxt` 58.20 MiB

### DISCUSSION THREAD

**Discussion by productdevbook** [nuxt/nuxt#25201] [General]
I think nuxt repo size is big.

nuxt go I think they are too much too big we need to have a look.

```
❯ git count-objects -vH

count: 0
size: 0 bytes
in-pack: 59898
packs: 1
size-pack: 58.20 MiB
prune-packable: 0
garbage: 0
size-garbage: 0 bytes
```

```
❯ git rev-list --objects --all | git cat-file --batch-check='%(objecttype) %(objectname) %(objectsize) %(rest)' | awk '/^blob/ && $3 > 1048576 {print $0}'

blob 53a282e439a38c102336ebd39720d3418f3d8160 2205218 docs/.yarn/releases/yarn-3.3.1.cjs
blob 47f24f66e01af080d1928a04b1bea50cf00ea21e 2202236 docs/.yarn/releases/yarn-3.3.0.cjs
blob 6369940602e486a105913ffcfb3d0757ba8a20a5 3210527 docs/public/social.png
blob 38f17fd49eada7cf53b9e2361ad4999c62aea373 2194904 docs/.yarn/releases/yarn-3.2.4.cjs
blob c038cf614d99b7ae0fd04a3daca5c9e6266850f4 1115977 docs/public/3D/gem.gltf
blob 12bde0351eaf8f1206fd8af67093a510755a0ad5 2173462 .yarn/releases/yarn-3.2.3.cjs
blob b3cadff6efb37a12712d12c2553ec703dbcaa4dd 2172670 docs/.yarn/releases/yarn-3.2.1.cjs
blob 55fe288f6413d1760ba1c1df888bf78050838220 5097983 docs/.yarn/releases/yarn-1.23.0-20210103.1434.cjs
blob 0912bea85eae19181704c80a91b74e74224adee8 2169057 .yarn/releases/yarn-3.2.2.cjs
blob 59267757f98a302a96d4ff8f5ccd3e48916a930a 2168472 .yarn/releases/yarn-3.2.0.cjs
blob f5f2adca83b2a46fc6a5709b43afd9eaaea3cbbe 2199165 .yarn/releases/yarn-3.1.1.cjs
blob 8584827cdbe5a4a90843575b78ae58e3e3292f83 2196921 .yarn/releases/yarn-3.1.0.cjs
blob 3c23463d23d6ed6e1cf8c90facaa143b84761c26 1980728 .yarn/releases/yarn-3.0.2.cjs
blob bf2e10dedd34ba77d10b202119e9eec864739c33 1979453 .yarn/releases/yarn-3.0.0.cjs
blob a15be7f95437977b2af2abc6b1880fc26b06b88b 1645622 .yarn/releases/yarn-2.4.2.cjs
blob 65ec5dde1973189c5c141a9a85a82f0c30320f6f 1640042 .yarn/releases/yarn-2.4.1.cjs
```

**Comment by rezansrv** [comment]
It looks like you've used the Git commands to inspect the size of objects in your Nuxt.js repository. The output shows various blobs (files) with their sizes. The sizes are in bytes, and you seem to be interested in files larger than 1 MB (1048576 bytes).

Here are some of the larger files in your repository:

1. `docs/.yarn/releases/yarn-3.3.1.cjs` - 2205218 bytes
2. `docs/.yarn/releases/yarn-3.3.0.cjs` - 2202236 bytes
3. `docs/public/social.png` - 3210527 bytes
4. `docs/.yarn/releases/yarn-3.2.4.cjs` - 2194904 bytes
5. `docs/public/3D/gem.gltf` - 1115977 bytes
6. `.yarn/releases/yarn-3.2.3.cjs` - 2173462 bytes
7. `docs/.yarn/releases/yarn-3.2.1.cjs` - 2172670 bytes
8. `docs/.yarn/releases/yarn-1.23.0-20210103.1434.cjs` - 5097983 bytes
9. `.yarn/releases/yarn-3.2.2.cjs` - 2169057 bytes
10. `.yarn/releases/yarn-3.2.0.cjs` - 2168472 bytes

If you are concerned about the overall size of your repository, you may want to consider the following:

1. **Unused Files:** Review if any of the larger files are unnecessary or if they can be optimized.

2. **Git History Cleanup:** If historical data is not essential, consider performing a Git history cleanup to remove large files that are no longer needed.

3. **Dependencies:** Check if any dependencies, especially the ones in the `.yarn/releases` directory, can be optimized or if there are unnecessary versions.

4. **Large Assets:** Assess if large assets, such as images or 3D models, can be optimized or if there are alternative ways to handle them.

Remember to be cautious when performing history cleanup or removing files, especially if your repository is shared or used collaboratively. Always make sure to have a backup or perform these actions in a controlled environment.

**Reply by productdevbook** [reply]
Thank you, I'm talking about the nuxt repo here.
